### ISSUE TITLE
[BUG] healthcheck fails due to redirect response (302) when using https

### ISSUE THREAD

**Issue by amcusack** [Lissy93/dashy#843]
### Environment

Self-Hosted (Docker)

### System

Debian GNU/Linux 11 (bullseye) / Docker version 20.10.5+dfsg1, build 55c4c88

### Version

Dashy version 2.1.1

### Describe the problem

I am running Dashy in a docker container on debian.

I specify my own ssl certificate as per instructions at [https://github.com/Lissy93/dashy/blob/master/docs/management.md#passing-a-self-signed-certificate-to-dashy](url)

dashy runs fine with https on port 443 , has correct certificate, no errors in log, able to configure & run with no problems. 

the only issue is the healthcheck..

its fails due to a redirect (302) message returned  - (redirection from http to https i assume)

the command used to test healthceck is: 
```bash
docker exec -it dashy yarn health-check
```
the result is : 
```bash
yarn run v1.22.15
$ node services/healthcheck
[Sat Aug 13 2022 12:10:24 GMT+0930 (GMT+09:30)] Running health check...
Status: 302
Request took 0.028 seconds
---
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Is there someway to get the healthcheck to test against https (443) instead of http (80)

my workaroud is to specify
 ```yml
healthcheck: 
   disable: true
``` 
in the docker-compose.yml but it would be nice to have healthcheck active.

Thanks.

### Please tick the boxes

- [X] You have explained the issue clearly, and included all relevant info
- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy
- [X] You've checked that this [issue hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide ![important](https://img.shields.io/badge/Important!-F6094E)
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by Lissy93** [comment]
Closing as dup of #840 and #768

**Comment by nwh3365** [comment]
I have been having the same issue.  I think the issue is with the isSsl test at the beginning of healthcheck.js.  It is determining if SSL is applicable strictly based on the SSL-related environment variables (SSL_PRIV_KEY_PATH and SSL_PUB_KEY_PATH) that are mentioned in the docs amcusack linked:

```
Passing a Self-Signed Certificate to Dashy
Once you've generated your SSL cert, you'll need to pass it to Dashy. This can be done by specifying the paths to your public and private keys using the SSL_PRIV_KEY_PATH and SSL_PUB_KEY_PATH environmental variables. Or if you're using Docker, then just pass public + private SSL keys in under /etc/ssl/certs/dashy-pub.pem and /etc/ssl/certs/dashy-priv.key respectively
```

The docs indicate that with Docker you just need to pass the key files to the container (without specifying the environment variables), and indeed Dashy will run in SSL mode by doing so.  However, because healthcheck.js is only using the environment variables to determine the SSL status, if you don't set them it assumes non-SSL and ultimately fails because it gets a "302 Found" instead of a "200 OK".

Ideally, the isSsl check should probably include whatever mechanism Dashy is using to put itself in SSL mode based on the presence of the key files in the container, however you can work around this issue simply by specifying the environment variables with docker run / docker compose:

  --env SSL_PUB_KEY_PATH="/etc/ssl/certs/dashy-pub.pem" \
  --env SSL_PRIV_KEY_PATH="/etc/ssl/certs/dashy-priv.key" \

By specifying the environment variables (in addition to passing the key files), the isSsl check determines the correct status and the rest of the healthcheck proceeds as it should.

This issue has been closed as a duplicate of #768 and #840, however it isn't resolved by those issues.  I'm going to create a new issue for it with the additional details, but wanted to post the workaround here for those that may find this issue (#843).
